Doc/whatsnew/3.15.rst

```diff
@@ -164,5 +164,20 @@ Deprecated C APIs
 Removed C APIs
 --------------
 
+*
```

Comment:
It's a little bit surprising to have a section (list) without test. Maybe write something like:

```
* Remove deprecated ``PyUnicode`` functions:
```